Gems/AzQtComponentsForPython/CMakeLists.txt

```diff
@@ -0,0 +1,224 @@
+
+set(PYSIDE2_SHARED_LIBRARIES "${PYSIDE2_LIB_DIR}/pyside2.abi3.lib")
+set(PYSIDE2_SHARED_LIBRARIES_D "${PYSIDE2_LIB_DIR}/pyside2_d.${LIB_EXTENSION}.lib")
+
+set(PYSIDE2_DLL "${PYSIDE2_LIB_DIR}/pyside2.abi3.dll")
+set(PYSIDE2_DLL_D "${PYSIDE2_LIB_DIR}/pyside2_d.${LIB_EXTENSION}.dll")
+  
+set(CMAKE_SKIP_BUILD_RPATH FALSE)
+set(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
+set(CMAKE_INSTALL_RPATH ${SHIBOKEN_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR})
+set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+
+# Define the shared library for which we will create bindings.
+set_property(TARGET ${AZQTCOMPONENTS_LIBRARY} PROPERTY PREFIX "")
+
+# Ensure the generated wrapper files have the correct header.
+set(LICENSE_FILE ${CMAKE_CURRENT_LIST_DIR}/license_header.txt)
+
+# Set up the options to pass to shiboken.
+set(shiboken_options --generator-set=shiboken --enable-parent-ctor-heuristic
+    --enable-pyside-extensions --enable-return-value-heuristic --use-isnull-as-nb_nonzero
+    --avoid-protected-hack --language-level=c++17
+    --license-file=${LICENSE_FILE}
+    ${INCLUDES}
+    -I${CMAKE_SOURCE_DIR}
+    -I${CMAKE_CURRENT_LIST_DIR}/..
+    -I"${LY_ROOT_FOLDER}/code/Framework/AzCore"
+    -I"${LY_ROOT_FOLDER}/code/Framework/AzQtComponents"
+    -I"${PROJECT_BINARY_DIR}/../AzQtComponents_autogen/include"
+    -I"${LY_ROOT_FOLDER}/code/Framework/AzCore/Platform/${PAL_PLATFORM_NAME}"
+    -I"${PYSIDE2_INCLUDE_DIR}"
+    -I"${SHIBOKEN_INCLUDE_DIR}"
+    -I"${PYSIDE2_INCLUDE_DIR}/QtWidgets"
+    -I"${PYSIDE2_INCLUDE_DIR}/QtCore"
+    -I"${PYSIDE2_INCLUDE_DIR}/QtGui"
+    -I"${PYSIDE2_TYPESYSTEM_PATH}"   
+    -I"${QT_MODULE_DIR}/include"
+    -I"${QT_MODULE_DIR}/include/QtWidgets"
+    -I"${QT_MODULE_DIR}/include/QtCore"
+    -I"${QT_MODULE_DIR}/include/QtGui"
+    -T"${PYSIDE2_PATH}/typesystems"
+    -T"${PYSIDE2_PATH}"
+    -T"${SHIBOKEN_INCLUDE_DIR}"
+    -T"${PYSIDE2_TYPESYSTEM_PATH}"
+    -T"${PYSIDE2_SHARE_PATH}"
+    --output-directory=${CMAKE_CURRENT_BINARY_DIR}
+)
+
+set(generated_sources_dependencies ${wrapped_header} ${typesystem_file})
+
+# Dummy library to run the custom shiboken command.
+ly_add_target(
+    NAME ${BINDINGS_LIBRARY}.bindings STATIC
+    NAMESPACE Gem
+    FILES_CMAKE
+        dummy_files.cmake
+    INCLUDE_DIRECTORIES
+        PRIVATE
+            ${CMAKE_CURRENT_LIST_DIR}/..
+            ${PYSIDE2_INCLUDE_DIR}
+    BUILD_DEPENDENCIES
+        PUBLIC
+            AZ::AzQtComponents
+            Gem::EditorPythonBindings.Static
+)
+
+# Add the custom shiboken command to the dummy library.
+add_custom_command(TARGET ${BINDINGS_LIBRARY}.bindings
+                    POST_BUILD
+                    COMMAND ${SHIBOKEN_GENERATOR_EXE_PATH} ${shiboken_options} ${wrapped_header} ${typesystem_file}
+                    DEPENDS ${generated_sources_dependencies} 
+                        Gem::AzQtComponentsForPython.WrappedFiles
+                    COMMENT "Running generator for ${typesystem_file}.")
```

Comment:
If anything, the logic to generate the bindings should either at the end of the CMake configure step or at the beginning the overall build and not as a POST_BUILD step of a placeholder target.

This also doesn't track the OUTPUT produced from the custom command(which the first signature of [add_custom_command](https://cmake.org/cmake/help/latest/command/add_custom_command.html#command:add_custom_command) support, so the command will not run if the bindings have been removed from the output directory.

If anything the OUTPUTS from the result of the Shiboken command should be listed as OUTPUT, so that
1. They are marked with the GENERATED property by CMake
2. CMake can track when they are output of date and properly track the dependencies

The [LyAutoGen](https://github.com/o3de/o3de/blob/development/cmake/LyAutoGen.cmake#L42-L48) custom command can be looked at as example of generating files at the beginning of the build step

Reply:
I've now redone this as an autogen step.